p2p/simulations/http_test.go

```diff
@@ -345,3 +384,129 @@ func TestHTTPNodeRPC(t *testing.T) {
+		}
+		nodes[i] = node
+	}
+	if err := client.ConnectNode(network.Id, nodes[0].ID, nodes[1].ID); err != nil {
+		t.Fatalf("error connecting nodes: %s", err)
+	}
+
+	// store some state in the test services
+	states := make([]string, nodeCount)
+	for i, node := range nodes {
+		rpc, err := client.RPCClient(context.Background(), network.Id, node.ID)
+		if err != nil {
+			t.Fatalf("error getting RPC client: %s", err)
+		}
+		defer rpc.Close()
+		state := fmt.Sprintf("%x", rand.Int())
+		if err := rpc.Call(nil, "test_setState", []byte(state)); err != nil {
+			t.Fatalf("error setting service state: %s", err)
+		}
+		states[i] = state
+	}
+
+	// create a snapshot
+	snap, err := client.CreateSnapshot(network.Id)
+	if err != nil {
+		t.Fatalf("error creating snapshot: %s", err)
+	}
+	for i, state := range states {
+		if string(snap.Nodes[i].Snapshot) != state {
+			t.Fatalf("expected snapshot state %q, got %q", state, snap.Nodes[i].Snapshot)
+		}
+	}
+
+	// create another network
+	network, err = client.CreateNetwork(&NetworkConfig{DefaultService: "test"})
+	if err != nil {
+		t.Fatalf("error creating network: %s", err)
+	}
+
+	// subscribe to events so we can check them later
+	events := make(chan *Event, 100)
+	sub, err := client.SubscribeNetwork(network.Id, events)
+	if err != nil {
+		t.Fatalf("error subscribing to network events: %s", err)
+	}
+	defer sub.Unsubscribe()
+
+	// load the snapshot
+	if err := client.LoadSnapshot(network.Id, snap); err != nil {
+		t.Fatalf("error loading snapshot: %s", err)
+	}
+
+	// check the nodes and connection exists
+	net, err := client.GetNetwork(network.Id)
+	if err != nil {
+		t.Fatalf("error getting network: %s", err)
+	}
+	if len(net.Nodes) != nodeCount {
+		t.Fatalf("expected network to have %d nodes, got %d", nodeCount, len(net.Nodes))
+	}
+	for i, node := range nodes {
+		id := net.Nodes[i].ID().String()
+		if id != node.ID {
+			t.Fatalf("expected node %d to have ID %s, got %s", i, node.ID, id)
+		}
+	}
+	if len(net.Conns) != 1 {
+		t.Fatalf("expected network to have 1 connection, got %d", len(net.Conns))
+	}
+	conn := net.Conns[0]
+	if conn.One.String() != nodes[0].ID {
+		t.Fatalf("expected connection to have one=%q, got one=%q", nodes[0].ID, conn.One)
+	}
+	if conn.Other.String() != nodes[1].ID {
+		t.Fatalf("expected connection to have other=%q, got other=%q", nodes[1].ID, conn.Other)
+	}
+
+	// check the node states were restored
+	for i, node := range nodes {
+		rpc, err := client.RPCClient(context.Background(), network.Id, node.ID)
+		if err != nil {
+			t.Fatalf("error getting RPC client: %s", err)
+		}
+		defer rpc.Close()
+		var state []byte
+		if err := rpc.Call(&state, "test_getState"); err != nil {
+			t.Fatalf("error getting service state: %s", err)
+		}
+		if string(state) != states[i] {
+			t.Fatalf("expected snapshot state %q, got %q", states[i], state)
+		}
+	}
+
+	// check we got all the events
+	x := &expectEvents{t, events, sub}
+	x.expect(
+		x.nodeEvent(nodes[0].ID, false),
```

Comment:
When loading a snapshot, we create and start each node one by one, and then restore the connections, and that is represented by the order here.